smartsim/_core/launcher/dragon/dragonLauncher.py

```diff
@@ -0,0 +1,362 @@
+from ...schemas import (
+    DragonBootstrapRequest,
+    DragonBootstrapResponse,
+    DragonHandshakeRequest,
+    DragonHandshakeResponse,
+    DragonRequest,
+    DragonRunRequest,
+    DragonRunResponse,
+    DragonStopRequest,
+    DragonStopResponse,
+    DragonUpdateStatusRequest,
+    DragonUpdateStatusResponse,
+)
+from ...utils.network import get_best_interface_and_address
+from ..launcher import WLMLauncher
+from ..step import DragonStep, LocalStep, Step
+from ..stepInfo import StepInfo
+
+logger = get_logger(__name__)
+
+
+class DragonLauncher(WLMLauncher):
+    """This class encapsulates the functionality needed
+    to launch jobs on systems that use Dragon on top of a workload manager.
+
+    All WLM launchers are capable of launching managed and unmanaged
+    jobs. Managed jobs are queried through interaction with with WLM,
+    in this case Dragon. Unmanaged jobs are held in the TaskManager
+    and are managed through references to their launching process ID
+    i.e. a psutil.Popen object
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._context = zmq.Context()
+        self._timeout = CONFIG.dragon_server_timeout
+        self._reconnect_timeout = CONFIG.dragon_server_reconnect_timeout
+        self._startup_timeout = CONFIG.dragon_server_startup_timeout
+        self._context.setsockopt(zmq.SNDTIMEO, value=self._timeout)
+        self._context.setsockopt(zmq.RCVTIMEO, value=self._timeout)
+        self._dragon_head_socket: t.Optional[zmq.Socket[t.Any]] = None
+        self._dragon_head_process: t.Optional[subprocess.Popen[bytes]]
+        self._comm_lock = RLock()
+
+    @property
+    def is_connected(self) -> bool:
+        return self._dragon_head_socket is not None
+
+    def _handsake(self, address: str) -> None:
+        self._dragon_head_socket = self._context.socket(zmq.REQ)
+        self._dragon_head_socket.connect(address)
+        request = DragonHandshakeRequest()
+        try:
+            response = self._send_request_as_json(request)
+            DragonHandshakeResponse.parse_obj(response)
+            logger.debug(
+                f"Successful handshake with Dragon server at address {address}"
+            )
+            return
+        except (zmq.ZMQError, zmq.Again) as e:
+            logger.debug(e)
+            self._dragon_head_socket.close()
+            self._dragon_head_socket = None
+            raise LauncherError(
+                f"Unsuccessful handshake with Dragon server at address {address}"
+            ) from e
+
+    def _set_timeout(self, timeout: int) -> None:
+        self._context.setsockopt(zmq.SNDTIMEO, value=timeout)
+        self._context.setsockopt(zmq.RCVTIMEO, value=timeout)
+
+    def connect_to_dragon(self, path: str) -> None:
+        with self._comm_lock:
+            # TODO use manager instead
+            if self.is_connected:
+                return
+
+            dragon_config_log = os.path.join(path, "dragon_config.log")
```

Comment:
Do you want to push the output file name into config.py::Config to make it easier to let other components touch it?

Reply:
Done done!